packages/stream-server/README.md

```diff
@@ -0,0 +1,56 @@
+# Meteor's DDP stream server based on SockJS
+
+> Network core of the Meteor's DDP.
+
+The `stream-server` package provides:
+
+* the default Meteor's stream server based on SockJS;
+* the mechanism to replace bundled `StreamServer`.
+
+## How to add your custom `StreamServer` to `StreamServers` and activate it
+
+For example your package has name `custom-stream-server`.
+
+1. Enable access to the `StreamServers` variable.
+
+    `custom-stream-server/package.js`
+
+    ```js
+    Package.onUse(function (api) {
+      // Add following lines
+      api.use(['stream-server'], 'server');
+      api.imply(['stream-server'], 'server');
+      api.export('StreamServers', 'server');
+      // ...
+    });
+    ```
+
+2. Push `CustomStreamServer` to the `StreamServers` array.
+
+    `custom-stream-server/server.js`
+
+    ```js
+    class CustomStreamServer {
+      constructor() {
+        // this.server = new WebSocket.Server();
+      }
+    }
+    
+    StreamServers.push(CustomStreamServer);
```

Comment:
In this example, let's recommend explicitly importing `StreamServers`:
```js
import { StreamServers } from "meteor/stream-server";
```
It might also be good to expose a function for adding new servers, instead of just an array, so that we have a hook for running initialization code after a new server is added.

Reply:
> It might also be good to expose a function for adding new servers, instead of just an array, so that we have a hook for running initialization code after a new server is added.

For this achievement we will need rewrite the `ddp-server/server_convenience.js` package.

```js
if (process.env.DDP_DEFAULT_CONNECTION_URL) {
  __meteor_runtime_config__.DDP_DEFAULT_CONNECTION_URL =
    process.env.DDP_DEFAULT_CONNECTION_URL;
}

Meteor.server = new Server;

Meteor.refresh = function (notification) {
  DDPServer._InvalidationCrossbar.fire(notification);
};

// Proxy the public methods of Meteor.server so they can
// be called directly on Meteor.
_.each(['publish', 'methods', 'call', 'apply', 'onConnection', 'onMessage'],
       function (name) {
         Meteor[name] = _.bind(Meteor.server[name], Meteor.server);
       });

// Meteor.server used to be called Meteor.default_server. Provide
// backcompat as a courtesy even though it was never documented.
// XXX COMPAT WITH 0.6.4
Meteor.default_server = Meteor.server;
```

I'm think the function for adding new servers is dangerous, because there is may be side effects in update the `Meteor.server` variable on the fly.

Reply:
> In this example, let's recommend explicitly importing StreamServers:
>
> ```js
> import { StreamServers } from "meteor/stream-server";
>```

@benjamn this part is done.